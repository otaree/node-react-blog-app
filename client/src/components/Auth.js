import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';

import Signup from './Signup';
import Login from './Login';

const Auth = () => (
    <section>
        <div>
            <ul>
                <li><Link to="/auth/signup">Sign Up</Link></li>
                <li><Link to="/auth/login">Login</Link></li>
            </ul>
        </div>
        <div>
            <Switch>
                <Route path="/auth/signup" component={Signup} />
                <Route path="/auth/login" component={Login} />
            </Switch>
        </div>
    </section>
);

export default Auth;