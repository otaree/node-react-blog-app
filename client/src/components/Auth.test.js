import React from 'react';
import { shallow } from 'enzyme';

import Auth from './Auth';

describe('Auth', () => {
    const auth = shallow(<Auth />);

    it('renders correctly', () => {
        expect(auth).toMatchSnapshot();
    });
});