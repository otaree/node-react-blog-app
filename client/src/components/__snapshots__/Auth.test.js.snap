// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Auth />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <ul>
                    <li>
                              <Link
                                        replace={false}
                                        to="/auth/signup"
                              >
                                        Sign Up
                              </Link>
                    </li>
                    <li>
                              <Link
                                        replace={false}
                                        to="/auth/login"
                              >
                                        Login
                              </Link>
                    </li>
          </ul>
</div>,
        <div>
          <Switch>
                    <Route
                              component={[Function]}
                              path="/auth/signup"
                    />
                    <Route
                              component={[Function]}
                              path="/auth/login"
                    />
          </Switch>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
                        <Link
                                    replace={false}
                                    to="/auth/signup"
                        >
                                    Sign Up
                        </Link>
            </li>
            <li>
                        <Link
                                    replace={false}
                                    to="/auth/login"
                        >
                                    Login
                        </Link>
            </li>
</ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                                replace={false}
                                to="/auth/signup"
                >
                                Sign Up
                </Link>
</li>,
              <li>
                <Link
                                replace={false}
                                to="/auth/login"
                >
                                Login
                </Link>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/auth/signup"
>
                  Sign Up
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Sign Up",
                  "replace": false,
                  "to": "/auth/signup",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/auth/login"
>
                  Login
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "replace": false,
                  "to": "/auth/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        path="/auth/signup"
            />
            <Route
                        component={[Function]}
                        path="/auth/login"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/auth/signup"
/>,
              <Route
                component={[Function]}
                path="/auth/login"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/auth/signup",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/auth/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <ul>
                        <li>
                                    <Link
                                                replace={false}
                                                to="/auth/signup"
                                    >
                                                Sign Up
                                    </Link>
                        </li>
                        <li>
                                    <Link
                                                replace={false}
                                                to="/auth/login"
                                    >
                                                Login
                                    </Link>
                        </li>
            </ul>
</div>,
          <div>
            <Switch>
                        <Route
                                    component={[Function]}
                                    path="/auth/signup"
                        />
                        <Route
                                    component={[Function]}
                                    path="/auth/login"
                        />
            </Switch>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                            <Link
                                          replace={false}
                                          to="/auth/signup"
                            >
                                          Sign Up
                            </Link>
              </li>
              <li>
                            <Link
                                          replace={false}
                                          to="/auth/login"
                            >
                                          Login
                            </Link>
              </li>
</ul>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Link
                                    replace={false}
                                    to="/auth/signup"
                  >
                                    Sign Up
                  </Link>
</li>,
                <li>
                  <Link
                                    replace={false}
                                    to="/auth/login"
                  >
                                    Login
                  </Link>
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/auth/signup"
>
                    Sign Up
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "replace": false,
                    "to": "/auth/signup",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/auth/login"
>
                    Login
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "replace": false,
                    "to": "/auth/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            path="/auth/signup"
              />
              <Route
                            component={[Function]}
                            path="/auth/login"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/auth/signup"
/>,
                <Route
                  component={[Function]}
                  path="/auth/login"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/auth/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/auth/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
